File: VB/NetCore/WhatIsMyBrowser.CommonTypesCoreVB/UserAgentParseDataRequestParseOptions.vb
Imports Newtonsoft.Json

Public Class UserAgentParseDataRequestParseOptions
    <JsonProperty("allow_servers_to_impersonate_devices", NullValueHandling:=NullValueHandling.Ignore)>
    Public Property AllowServersToImpersonateDevices As Nullable(Of Boolean)

    <JsonProperty("return_metadata_for_useragent", NullValueHandling:=NullValueHandling.Ignore)>
    Public Property ReturnMetadataForUserAgent As Nullable(Of Boolean)

    <JsonProperty("dont_sanitize", NullValueHandling:=NullValueHandling.Ignore)>
    Public Property DoNotSanitize As Nullable(Of Boolean)
End Class

File: CS/NetCore/WhatIsMyBrowser.CommonTypesCore/UserAgentParseDataRequestParseOptions.cs
using Newtonsoft.Json;

namespace WhatIsMyBrowser.CommonTypesCore
{
    public class UserAgentParseDataRequestParseOptions
    {
        [JsonProperty("allow_servers_to_impersonate_devices", NullValueHandling = NullValueHandling.Ignore)]
        public bool? AllowServersToImpersonateDevices { get; set; }

        [JsonProperty("return_metadata_for_useragent", NullValueHandling = NullValueHandling.Ignore)]
        public bool? ReturnMetadataForUserAgent { get; set; }

        [JsonProperty("dont_sanitize", NullValueHandling = NullValueHandling.Ignore)]
        public bool? DoNotSanitize { get; set; }
    }
}